src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -99,8 +99,9 @@ default void prepareTmpFldr(FilePath tmpFldr, String runnerScript) throws IOExce
         zipFileLocation.unzip(tmpFldr);
     }
     
-    default String getRunnerScript(String script, String params) {
+    default String getRunnerScript(String script, String params, String uniqueTmpFldrName) {
         script = script.replace("${PARAMS}", params);
+        script = script.replaceAll("tmp_f", uniqueTmpFldrName);

```

Comment:
Why two different approaches to placeholder strings?
```suggestion
        script = script.replace("${PARAMS}", params);
        script = script.replaceAll("${TMPDIR}", uniqueTmpFldrName);

```

Reply:
tried this approach already however the occurrence $ is more than once i guess Java API .replaceAll throws an exception in its subsequent occurrence

Reply:
Looks like `replaceAll` takes a regular expression, not just a raw string value. $ has meaning in regexp. You should be able to escape it with backslash. Note { also has special meaning in regexp, so you'll need to escape that as well.

Reply:
Done